Dockerfile

```diff
@@ -415,6 +415,85 @@ ENV PATH="/usr/libexec/rust/bin:$PATH" LD_LIBRARY_PATH="/usr/libexec/rust/lib"
 
 # =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+FROM sdk as sdk-grub
+
+USER root
+ARG HOST_ARCH
+ENV GRUB_VER="2.06-61.amzn2023.0.9"
+
+RUN \
+  mkdir -p /usr/libexec/tools /usr/share/licenses/grub && \
+  chown -R builder:builder /usr/libexec/tools /usr/share/licenses/grub
+
+USER builder
+WORKDIR /home/builder
+COPY ./hashes/grub /home/builder/hashes
+COPY ./patches/grub /home/builder/patches
+
+# This rather elaborate way of unpacking the sources and applying the
+# patches mimics the way GRUB is built in Bottlerocket, which in turn
+# mimics the way that Amazon Linux and Fedora build it.
+RUN \
+  sdk-fetch /home/builder/hashes && \
```

Comment:
would it be worth having this as a script instead of a long  "one-liner"?

Reply:
> Would it be worth having this as a script instead of a long "one-liner"?

The line I am drawing in my head:
* for builds that we only do once for the host arch, it's more idiomatic to do them inline in the Dockerfile
* for builds that we need to do for multiple versions or target architectures, it's better to do them in a script

At some point I could see converging on the second approach for everything, but it does make the Dockerfile more abstract and less useful as a way to immediately see how something is built.